src/index.js

```diff
@@ -153,6 +151,40 @@ const { minify: minifyWorker } = require("./minify");
 
 const warningRegex = /\s.+:+([0-9]+):+([0-9]+)/;
 
+/**
+ * @template T
+ * @param fn {(function(): any) | undefined}
```

Comment:
shouldn't this be `@param {function (): T} fn` instead ? This would allow typescript to infer `T` properly. Also, the argument should not be undefined as that would break. The current signature means that type checking won't catch mistakes (the need to override the type when using the variable hides the fact that the type checker cannot trust it)